### ISSUE TITLE
Instructions on how to install

### ISSUE THREAD

**Issue by wcbonner** [tmobile/DevEdge-IoTDevKit-ZephyrSDK#38]
My device arrived this week, but I've not been able to get it to talk to the web application. 

My device reports:
```
uart:~$ tmo version
Built for: tmo_dev_edge
Zephyr:    88ca3aca98c0788e2c45db5c3fb9c8523293a8c0
TMO RTOS:  zephyr-v3.1.0-1333-g7dd7f70747
TMO SDK:   tags/tmz-sdk-1.13.2-qa-0-gb3a17c5f4e
```
https://devedge.t-mobile.com/support/iot-developer-kit/release-notes seems to say that https://github.com/tmobile/DevEdge-IoTDevKit-ZephyrSDK/releases/tag/tmz-sdk-1.15.0-qa should be the first public release and I'm wondering if the problems I have are related to advancements since the version I'm using.

Is building the Zephyr environment on my local machine following https://github.com/tmobile/DevEdge-IoTDevKit-ZephyrSDK/tree/tmz-sdk-1.15.0-qa/samples/tmo_shell the only way to update the machine right now?

**Comment by asanchezdelc** [comment]
Hi 👋 , I'm also facing the same issue. The json demo app returns an error when making HTTP request to: `https://devkitmqtt.devedge.t-mobile.com/prd/bridge/` Are there any instructions on upgrading the device?

**Comment by wcbonner** [comment]
The release notes link seems to have changed from the old:

> https://devedge.t-mobile.com/support/iot-developer-kit/release-notes seems to say that 

to this new link: https://devedge.t-mobile.com/documentation-hub/iot-developer-kit/release-notes

It didn't look like the description of the most recent release has changed, but I suppose I'll now need to go through all those pages and see if it's added anything about updating the system.

**Comment by wcbonner** [comment]
further information:

```
uart:~$ tmo json iface 2
uart:~$ tmo json enable
uart:~$ server_url: https://devkitmqtt.devedge.t-mobile.com/prd/bridge/LoFWui6a
payload:
{
"accelerometer":{
"x":0.19,
"y":-0.61,
"z":9.40
},
"battery":{
"voltage":0.000,
"percent":0,
"state":"not-attached"
},
"cellSignalStrength":{
"dbm":-79
},
"temperature":{
"temperatureCelsius":25.2
},
"ambientLight":{
"visibleLux":43.78,
"irLux":10.37
},
"pressure":{
"kPa":101.63
},
"map":{
"lat":0.000000,
"lng":0.000000,
"alt":0.00,
"hdop":0.00
}
}

Error connecting socket, error: -1, errno: 113
tmo json disable
uart:~$ tmo version
Built for: tmo_dev_edge
Zephyr:    88ca3aca98c0788e2c45db5c3fb9c8523293a8c0
TMO RTOS:  zephyr-v3.1.0-1333-g7dd7f70747
TMO SDK:   tags/tmz-sdk-1.13.2-qa-0-gb3a17c5f4e
uart:~$
```

**Comment by asanchezdelc** [comment]
@wcbonner I get the same error. Although is a valid URL, because I can HTTP post to it with the payload and the web app receives the values. 

```shell
curl --request POST \
  --url https://devkitmqtt.devedge.t-mobile.com/prd/bridge/CODE \
  --header 'Content-Type: application/json' \
  --data '{
"accelerometer":{
"x":0.17,
"y":-0.11,
"z":9.96
},
"battery":{
"voltage":3.993,
"percent":66,
"state":"charging"
},
"cellSignalStrength":{
"dbm":-120
},
"temperature":{
"temperatureCelsius":37.7
},
"ambientLight":{
"visibleLux":13.45,
"irLux":2.90
},
"pressure":{
"kPa":98.70
},
"map":{
"lat":0.000000,
"lng":0.000000,
"alt":0.00,
"hdop":0.00
}
}'
```

**Comment by wcbonner** [comment]
I just tried to build the project on my Raspberry Pi4 running Raspian Bullseye. After issuing all of these commands I ended up with a simple CMake version error saying **CMake 3.20.0 or higher is required.  You are running version 3.18.4**.

I'd also seen some errors scrolling past during the west operation that indicated that many parts may be oriented towards win32 platform for development.

```
sudo apt install --no-install-recommends git cmake ninja-build gperf   ccache dfu-util device-tree-compiler wget   python3-dev python3-pip python3-setuptools python3-tk python3-wheel xz-utils file make gcc libsdl2-dev libmagic1
pip3 install --user -U west
echo 'export PATH=~/.local/bin:"$PATH"' >> ~/.bashrc
source ~/.bashrc
mkdir ~/zephyrproject
cd ~/zephyrproject
git clone https://github.com/tmobile/DevEdge-IoTDevKit-ZephyrRTOS zephyr
git clone https://github.com/tmobile/DevEdge-IoTDevKit-ZephyrSDK tmo-zephyr-sdk
west update
west init ~/zephyrproject
west update
west zephyr-export
pip3 install --user -r ~/zephyrproject/zephyr/scripts/requirements.txt
west build ~/zephyrproject/tmo-zephyr-sdk/samples/tmo_shell -p -b tmo_dev_edge -- -DBOARD_ROOT=/home/wim/zephyrproject/tmo-zephyr-sdk
```

**Comment by johnlange2** [comment]
Hi all, the issue being discussed here may be due to out of date 9116 (WiFi/BLE chip) firmware. To verify whether this is the issue, run the command `tmo dfu version 2`. If it shows 1610.2.6.0.0.34, it is correct. If it shows 1610.2.4.0.0.36, then the 9116 firmware needs to be updated (to 1610.2.6.0.0.34). We are currently working on streamlined instructions on how to update the 9116 firmware, and i'll post back here with the link once it's ready.

Note that you can also test the web app by using the onboard modem assuming it is connected. To check whether your modem is connected, run `tmo modem 1 conn_sts`. It should show T-Mobile. To use the modem for the web app, first run `tmo json iface 1`. Then run `tmo json enable`.

**Comment by wcbonner** [comment]
@johnlange2 I'm looking forward to seeing instructions on how to update the platform. Hopefully when that information gets published it'll be mentioned here. 

```
uart:~$ tmo dfu version 2
cmd_dfu_get_version: target: 2

RS9116W FW version is: 1610.2.4.0.0.36
The current RS9116W FW version is: 1610.2.4.0.0.36

uart:~$ tmo dfu version 1
cmd_dfu_get_version: target: 1

The current Murata 1SC FW version is RK_03_02_00_00_20161_001
```

**Comment by johnlange2** [comment]
Instructions on how to update the RS9116W firmware can be found [here](https://devedge.t-mobile.com/documentation-hub/iot-developer-kit/upgrading-your-iot-developer-kits-rs9116w-firmware)

**Comment by wcbonner** [comment]
> Instructions on how to update the RS9116W firmware can be found [here](https://devedge.t-mobile.com/documentation-hub/iot-developer-kit/upgrading-your-iot-developer-kits-rs9116w-firmware)

@johnlange2 Thanks. I was able to get it to update following those instructions and now it connects to the web application. My device isn't reporting gps coordinates. Is that because the device is not as sensitive as the other gps I've got in my apartment, or is the current firmware not reporting it?

New version information:
```
uart:~$ tmo version
Built for: tmo_dev_edge
Zephyr:    88ca3aca98c0788e2c45db5c3fb9c8523293a8c0
TMO RTOS:  zephyr-v3.1.0-1333-g7dd7f70747
TMO SDK:   tags/tmz-sdk-1.13.2-qa-0-gb3a17c5f4e
uart:~$ tmo dfu version
Missing required arguments
Usage: tmo dfu version <target>
       target: 0 for mcu, 1 for modem, 2 for wifi
uart:~$ tmo dfu version 1
cmd_dfu_get_version: target: 1

The current Murata 1SC FW version is RK_03_02_00_00_20161_001
uart:~$ tmo dfu version 2
cmd_dfu_get_version: target: 2

RS9116W FW version is: 1610.2.6.0.0.34
The current RS9116W FW version is: 1610.2.6.0.0.34
```
And JSON data being submitted:
```
[00:06:24.089,000] <inf> tmo_http_request: Response status code: 200, OK
uart:~$ server_url: https://devkitmqtt.devedge.t-mobile.com/prd/bridge/
payload:
{
"accelerometer":{
"x":0.21,
"y":-0.65,
"z":9.40
},
"battery":{
"voltage":0.000,
"percent":0,
"state":"not-attached"
},
"cellSignalStrength":{
"dbm":-97
},
"temperature":{
"temperatureCelsius":25.8
},
"ambientLight":{
"visibleLux":243.22,
"irLux":23.75
},
"pressure":{
"kPa":102.69
},
"map":{
"lat":0.000000,
"lng":0.000000,
"alt":0.00,
"hdop":0.00
}
}

Sending request...
http_client_req returned 484
[00:06:35.420,000] <inf> tmo_http_request: Response status code: 200, OK
uart:~$ tmo json disable
```

**Comment by johnlange2** [comment]
@wcbonner Glad to hear that you were able to update your RS9116W!

The most likely cause of not getting a GPS fix is that the signal isn't strong enough indoors. You can sometimes get fixes better when the dev kit is placed near a window, but it's really best to try it outdoors with full view of the sky. We have some enhancements planned to improve GNSS performance and also get location while indoors.

Just to verify that the GNSS firmware is loaded properly, you can run `tmo gnssversion`. It should print a string that isn't blank or all zeros.

**Comment by asanchezdelc** [comment]
@johnlange2 Thanks for the detailed instructions. I was able to upgrade and everything seems to work properly.

**Comment by jcdang** [comment]
@johnlange2 I think the lat/lon reporting might be a regression.  I definitely remember seeing my house before the firmware update.  Was it using some sort of geoip service before?

**Comment by rickey318** [comment]
I have followed the update in the provided link. And it updates. But, when I do the: uart:~$ fs cat /tmo/aws_session.txt
I always get: Failed to obtain file /tmo/aws_session.txt (err: -2)

Is there something missing on mines? I did the reset with the button before trying and it is connected to my wifi.

**Comment by johnlange2** [comment]
@jcdang 
> Was it using some sort of geoip service before?

No, we haven't used external location services in any release to date.

**Comment by johnlange2** [comment]
@rickey318 
> Is there something missing on mines?
Yes, from your description, it appears this file is missing, and it is needed to use the web app. I'll see if i can find a solution for you.

**Comment by sihekuang** [comment]
@rickey318 your situation requires some manual intervention on our side. Please email your device's IMEI with `tmo modem 1 imei` command to [support@devedge.com](mailto:support@devedge.com). One of our teammates will reach out to you to get the problem resolved. Thank you!

**Comment by rickey318** [comment]
> @rickey318 your situation requires some manual intervention on our side. Please email your device's IMEI with `tmo modem 1 imei` command to [support@devedge.com](mailto:support@devedge.com). One of our teammates will reach out to you to get the problem resolved. Thank you!

Hello, I have submitted that information to the support email. Thank you.

**Comment by wcbonner** [comment]
I'm going to close this issue because the updated instructions at the main site got my device running the correct version and working with the web app. 

@rickey318 If you still have an open issue, I suggest opening a separate issue. 

Wim
